internal/handlers/common/projection.go

```diff
@@ -205,57 +276,195 @@ func applyComplexProjection(k1 string, doc, projectionVal *types.Document) (err
 				return nil
 			}
 			must.NoError(doc.Set(k1, res))
+
 		default:
-			return NewError(ErrCommandNotFound,
-				lazyerrors.Errorf("applyComplexProjection: unknown projection operator: %q", projectionType),
-			)
+			return NewErrorMsg(ErrCommandNotFound, projectionName+" not supported")
 		}
 	}
-
-	return
+	return err
 }
 
-// filterFieldArrayElemMatch is for elemMatch conditions.
-func filterFieldArrayElemMatch(k1 string, doc, conditions *types.Document, docValueArray *types.Array) (found int, err error) {
-	for k2ConditionField, conditionValue := range conditions.Map() {
-		switch elemMatchFieldCondition := conditionValue.(type) {
-		case *types.Document: // TODO field2: { $gte: 10 }
+func findInArray(k1, k2 string, value any, doc *types.Document, compareRes []types.CompareResult) bool {
+	docValueArray := must.NotFail(doc.GetByPath(k1)).(*types.Array)
 
-		case *types.Array:
-			panic("unexpected code")
+	found := -1
+fieldArray:
+	for j := 0; j < docValueArray.Len(); j++ {
+		e, err := docValueArray.Get(j)
+		if err != nil {
+			continue
+		}
 
-		default: // field2: value
-			found = -1 // >= 0 means found
+		if found >= 0 {
+			doc.RemoveByPath(k1, strconv.Itoa(j))
+			j -= 1
+			continue
+		}
+		switch e := e.(type) {
+		case *types.Document:
+			var d any
+			d, err = e.Get(k2)
+			if err != nil {
+				doc.RemoveByPath(k1, strconv.Itoa(j))
+				j -= 1
+				continue
+			}
+			switch value := value.(type) {
+			// TODO https://github.com/FerretDB/FerretDB/issues/439 resolve in nested
```

Comment:
That's the issue this PR resolves

Reply:
no, it does not resolve nested.

Reply:
now it is resolved, but I think here is one another test case is necessary.